File: python/guessing_game.py
import random


secret_number = random.randint(1, 100)

while True:
  try:
    guess = int(input("Guess the number: "))
  except:
    print("please type a number!")
    continue
  
  print(f"You guessed {guess}")

  if (guess > secret_number):
    print("Too big!")
  elif (guess < secret_number):
    print("Too small!")
  else:
    print("You Won!")
    break

File: c++/guessing_game.cpp
#include <cstdlib>
#include <iostream>
#include <string> 
using namespace std;

int main() {
  srand(time(0));
  int secret = rand() % 100;

  string guess;
  int num_guess;
  while (true) {
    cout << "guess a number: " << endl;
    
    cin >> guess;

    try {
      num_guess = stoi(guess);
    } catch (exception error) {
      cout << "please type a number!" << endl;
      continue;
    }

    cout << "you guess:" << guess << endl;
    if (num_guess > secret) cout << "Too Big!" << endl;
    else if (num_guess < secret) cout << "Too Small!" << endl;
    else {
      cout << "Too Won!" << endl;
      break;
    }
  }
}
